cache.go

```diff
@@ -325,19 +335,36 @@ func (c *Cache[K, V]) GetOrSet(key K, value V, opts ...Option[K, V]) (*Item[K, V
 
 	applyOptions(&setOpts, opts...)
 
-	return c.Set(key, value, setOpts.ttl), false
+	item := c.set(key, value, setOpts.ttl)
+	c.items.mu.Unlock()
+
+	return item, false
 }
 
 // GetAndDelete deletes the value for a key, returning the previous
 // value if any. The retrieved result reports whether the key was present.
+// If a loader is defined, the value will be loaded during get,
+// but will state, that the result is not present in the cache.
```

Comment:
If the value is loaded from the loader and is non-nil, the bool return value should be set to `true`. Anything other than that would be a breaking change at the moment and might lead to some pretty confusing situations.

Reply:
And the doc states, `The retrieved result reports whether the key was present,` not `The retrieved result reports whether the key was deleted`.

Reply:
The line 361 could look like this:
```go
item := getOpts.loader.Load(c, key)
return item, item != nil
```